File: 2838/2838.go
func maximumCoinsHeroesCanCollect(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := maximumCoinsHeroesCanCollect(root.Left)
	r := maximumCoinsHeroesCanCollect(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 2838/2838.c
int maximumCoinsHeroesCanCollect(struct TreeNode* root) {
    if (!root) return 0;
    int l = maximumCoinsHeroesCanCollect(root->left);
    int r = maximumCoinsHeroesCanCollect(root->right);
    return 1 + (l > r ? l : r);
}